### ISSUE TITLE
RubyRipper doesn't work well on systems with multiple cdroms

### ISSUE THREAD

**Issue by GoogleCodeExporter** [bleskodev/rubyripper#50]
```
I use RubyRipper 0.3, downloaded from RubyForge, installed on my Kubuntu 
6.10. It works well ripping discs from my first
cdrom (/dev/hdc) but not when using the second one (dev/hdd). This is a 
quite extensive bug report but I found it necessary
to enter 'verbose mode' in order to explain it all. Also, I am a Linux 
noob, so please bear with me.

First, a look at my system. This is my /etc/fstab:

+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
/home/effo> cat /etc/fstab
# /etc/fstab: static file system information.
#
# <file system> <mount point>   <type>  <options>       <dump>  <pass>
proc            /proc           proc    defaults        0       0
# /dev/hdb2 -- converted during upgrade to edgy
UUID=c5ca0f83-6383-48ff-8536-ec358819dadb / ext3 defaults,errors=remount-
ro 0 1
# /dev/hdb3 -- converted during upgrade to edgy
UUID=487c0d17-b63e-4d79-8e17-2fc7007ade7c /home ext3 defaults 0 2
# /dev/hdd1 -- converted during upgrade to edgy
#UUID=DC589D85589D5ED8 /media/hdd1 ntfs defaults,nls=utf8,umask=007,gid=46 
0 1
# /dev/hdb4 -- converted during upgrade to edgy
UUID=8413aef5-8feb-4cf1-a476-93b90933191a none swap sw 0 0
/dev/hdc        /media/cdrom0   udf,iso9660 user,noauto     0       0
/dev/hdd        /media/cdrom1   udf,iso9660 user,noauto     0       0
/dev/fd0        /media/floppy0  auto    rw,user,noauto  0       0
/dev/hdb5       /media/transfer vfat    
user,utf8,fmask=0077,dmask=0077,uid=1000,gid=1000  0   0
/dev/hdb6       /media/mp3-1    vfat    
user,utf8,fmask=0077,dmask=0077,uid=1000,gid=1000  0   0
/dev/hdb7       /media/mp3-2    vfat    
user,utf8,fmask=0077,dmask=0077,uid=1000,gid=1000  0   0
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

As you see, I have two CDs, /dev/hdc and /dev/hdd

Now a look at /cdrom:

+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
/home/effo> ls -al /cdro*
lrwxrwxrwx 1 root root 11 2006-11-02 06:38 /cdrom -> media/cdrom
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

So the default /cdrom points to /media/cdrom. Let's look at /media too:

+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
/home/effo> ls -al /media/ | grep cdro*
lrwxrwxrwx  1 root root     6 2006-11-02 06:38 cdrom -> cdrom0
drwxr-xr-x  2 root root  4096 2006-11-02 06:38 cdrom0
drwxr-xr-x  2 root root  4096 2007-01-15 10:06 cdrom1
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

So /cdrom is linked to /media/cdrom which is linked to /media/cdrom0. 
There is also a /media/cdrom1.

Now use cdparanoia to get some info about the cdroms. With no disc in any 
of the drives I get the following result:

+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
/home/effo> cdparanoia -vsQ
cdparanoia III release 9.8 (March 23, 2001)
(C) 2001 Monty <monty@xiph.org> and Xiphophorus
FreeBSD porting (c) 2003
        Simon 'corecode' Schubert <corecode@corecode.ath.cx>

Report bugs to paranoia@xiph.org
http://www.xiph.org/paranoia/

Checking /dev/cdrom for cdrom...
        Testing /dev/cdrom for cooked ioctl() interface
                CDROM sensed: ATAPI compatible _NEC DVD_RW ND-3540A

004: Unable to read table of contents header

Unable to open disc.  Is there an audio CD in the drive?
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

Seem like cdparanoia by default used /dev/cdrom for querying. Let's use 
cdparanoia while specifying the drive:

+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
home/effo> cdparanoia -vsQ -d /dev/hdc
cdparanoia III release 9.8 (March 23, 2001)
(C) 2001 Monty <monty@xiph.org> and Xiphophorus
FreeBSD porting (c) 2003
        Simon 'corecode' Schubert <corecode@corecode.ath.cx>

Report bugs to paranoia@xiph.org
http://www.xiph.org/paranoia/

Checking /dev/hdc for cdrom...
        Testing /dev/hdc for cooked ioctl() interface
                CDROM sensed: ATAPI compatible _NEC DVD_RW ND-3540A

004: Unable to read table of contents header

Unable to open disc.  Is there an audio CD in the drive?
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

Pretty much the same result. At least we now know that /dev/hdc is a NEC 
DVD_RW. Let's check the other drive as well:

+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
/home/effo> cdparanoia -vsQ -d /dev/hdd
cdparanoia III release 9.8 (March 23, 2001)
(C) 2001 Monty <monty@xiph.org> and Xiphophorus
FreeBSD porting (c) 2003
        Simon 'corecode' Schubert <corecode@corecode.ath.cx>

Report bugs to paranoia@xiph.org
http://www.xiph.org/paranoia/

Checking /dev/hdd for cdrom...
        Testing /dev/hdd for cooked ioctl() interface
                CDROM sensed: ATAPI compatible HL-DT-ST GCE-8320B

004: Unable to read table of contents header

Unable to open disc.  Is there an audio CD in the drive?
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

Not much to say here, except we now know that /dev/hdd is a HL-DT-ST GCE-
8320B
Finally, this is the .rubyripper_settings file I start off with:

+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
wav=true
flacsettings=--best
mp3settings=--preset fast standard
vorbissettings=-q 6
rippersettings=-v
othersettings=
freedb=true
hostname=my_secret.com
username=anonymous
site=freedb.org
cdrom=/dev/hdc
offset=0
destination_dir=/home/effo/temp/%a - %b/%n - %t
max_tries=0.0
req_matches_all=2.0
req_matches_errors=2.0
temp_dir=/tmp/rr/
verbose=true
playlist=true
debug=true
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

Let's insert a two track cd single in /dev/hdc. cdparanoia now outputs the 
following:

+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
/home/effo> cdparanoia -vsQ -d /dev/hdc
cdparanoia III release 9.8 (March 23, 2001)
(C) 2001 Monty <monty@xiph.org> and Xiphophorus
FreeBSD porting (c) 2003
        Simon 'corecode' Schubert <corecode@corecode.ath.cx>

Report bugs to paranoia@xiph.org
http://www.xiph.org/paranoia/

Checking /dev/hdc for cdrom...
        Testing /dev/hdc for cooked ioctl() interface
                CDROM sensed: ATAPI compatible _NEC DVD_RW ND-3540A

Verifying drive can read CDDA...
        Expected command set reads OK.

Table of contents (audio tracks only):
track        length               begin        copy pre ch
===========================================================
  1.    25150 [05:35.25]        0 [00:00.00]    no   no  2
  2.    23137 [05:08.37]    25150 [05:35.25]    no   no  2
TOTAL   48287 [10:43.62]    (audio only)
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

So there really is a two-track disc in /dev/hdc. Now I can use rrip_cli to 
rip one of the tracks. Fine.

Now eject the disc from /dev/hdc and insert a full lenght album in /dev/
hdd. Then run cdparanoia to verify the album
is there:

+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
home/effo> cdparanoia -vsQ -d /dev/hdd
cdparanoia III release 9.8 (March 23, 2001)
(C) 2001 Monty <monty@xiph.org> and Xiphophorus
FreeBSD porting (c) 2003
        Simon 'corecode' Schubert <corecode@corecode.ath.cx>

Report bugs to paranoia@xiph.org
http://www.xiph.org/paranoia/

Checking /dev/hdd for cdrom...
        Testing /dev/hdd for cooked ioctl() interface
                CDROM sensed: ATAPI compatible HL-DT-ST GCE-8320B

Verifying drive can read CDDA...
        Expected command set reads OK.

Table of contents (audio tracks only):
track        length               begin        copy pre ch
===========================================================
  1.    17715 [03:56.15]        0 [00:00.00]    no   no  2
  2.    20690 [04:35.65]    17715 [03:56.15]    no   no  2
  3.    17527 [03:53.52]    38405 [08:32.05]    no   no  2
  4.    26620 [05:54.70]    55932 [12:25.57]    no   no  2
  5.    11923 [02:38.73]    82552 [18:20.52]    no   no  2
  6.    14777 [03:17.02]    94475 [20:59.50]    no   no  2
  7.    20893 [04:38.43]   109252 [24:16.52]    no   no  2
  8.    17062 [03:47.37]   130145 [28:55.20]    no   no  2
  9.    15130 [03:21.55]   147207 [32:42.57]    no   no  2
 10.    23265 [05:10.15]   162337 [36:04.37]    no   no  2
 11.    73848 [16:24.48]   185602 [41:14.52]    no   no  2
TOTAL  259450 [57:39.25]    (audio only)
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

Looks fine so let's update the .rubyripper_settings file to use /dev/hdd 
and start the cli

+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
/home/effo> sed s/hdc/hdd/ .rubyripper_settings
/home/effo> rrip_cli
No audio disc found in /dev/hdd

Do you want to change your settings? (y/n) : n
No audio-disc found, exiting...
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

What? No disc? But cdparanoia told me there IS a disc in /dev/hdd! Let's 
check it again:

+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
/home/effo> cdparanoia -vsQ -d /dev/hdd
cdparanoia III release 9.8 (March 23, 2001)
(C) 2001 Monty <monty@xiph.org> and Xiphophorus
FreeBSD porting (c) 2003
        Simon 'corecode' Schubert <corecode@corecode.ath.cx>

Report bugs to paranoia@xiph.org
http://www.xiph.org/paranoia/

Checking /dev/hdd for cdrom...
        Testing /dev/hdd for cooked ioctl() interface
                CDROM sensed: ATAPI compatible HL-DT-ST GCE-8320B

Verifying drive can read CDDA...
        Expected command set reads OK.

Table of contents (audio tracks only):
track        length               begin        copy pre ch
===========================================================
  1.    17715 [03:56.15]        0 [00:00.00]    no   no  2
  2.    20690 [04:35.65]    17715 [03:56.15]    no   no  2
  3.    17527 [03:53.52]    38405 [08:32.05]    no   no  2
  4.    26620 [05:54.70]    55932 [12:25.57]    no   no  2
  5.    11923 [02:38.73]    82552 [18:20.52]    no   no  2
  6.    14777 [03:17.02]    94475 [20:59.50]    no   no  2
  7.    20893 [04:38.43]   109252 [24:16.52]    no   no  2
  8.    17062 [03:47.37]   130145 [28:55.20]    no   no  2
  9.    15130 [03:21.55]   147207 [32:42.57]    no   no  2
 10.    23265 [05:10.15]   162337 [36:04.37]    no   no  2
 11.    73848 [16:24.48]   185602 [41:14.52]    no   no  2
TOTAL  259450 [57:39.25]    (audio only)
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

Yupp, the disc is there. So even if there is a device specified in
 .rubyripper_settings, it seems like RubyRipper tries
to use /dev/hdc (I guess that is caused by following /cdrom which is 
linked to /media/cdrom which is linked to /media/cdrom0
which is where /dev/hdc is mounted). There is also another sign indicating 
that /cdrom is used instead of the specified
device. Let's insert the cd single in /dev/hdc again, with the album still 
in /dev/hdd. RubuRipper is still configured
to use /dev/hdd:

+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
/home/effo> cat .rubyripper_settings | grep dev
cdrom=/dev/hdd
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

Now start rrip_cli:

+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
/home/effo> rrip_cli
Audio-disc found, number of tracks : 2, total playlength : 10:43
Fetching freedb info...

FREEDB INFO
1) Artist : Stefan Sundström
2) Album : Fabler Från Bällingebro
3) Genre : Folk/Rock
4) Year : 2006
5) Track 1 : Gässen På Fälten
6) Track 2 : Haren


Should all tracks be ripped ? (y/n)            
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

This is VERY interessting. The application starts, but with a strange 
result. The information from FREEDB is for the
record in /dev/hdd but it seem like the disc layout - with only two tracks 
- is read from /dev/hdc (or /cdrom). So with
a disc in /dev/hdc it is possible to start the application and read a disc 
from /dev/hdd, but it will still use the
layout read from /dev/hdc.

To summarize: It seems like RubyRipper will use the device that /cdrom 
poinst to, no matter if anohter device is specified
in the .rubyripper_settings file
```

Original issue reported on code.google.com by `crazysw...@gmail.com` on 21 Jan 2007 at 9:07
